core/src/gl/texture.cpp

```diff
@@ -189,6 +191,8 @@ void Texture::generate(GLuint _textureUnit) {
     GL_CHECK(glTexParameteri(m_target, GL_TEXTURE_WRAP_T, m_options.wrapping.wrapt));
 
     m_generation = RenderState::generation();
+
+    m_mainThreadJobQueue.makeCurrentThreadTarget();
```

Comment:
`makeCurrentThreadTarget()` sets the current thread as the thread that any queued jobs will be run on later. It doesn't have anything directly to do with GL.